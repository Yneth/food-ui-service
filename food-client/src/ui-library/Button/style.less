@import "../styles/variables";

.button {
  display: inline-block;

  border-width: 2px;
  border-style: solid;
  border-radius: 3px;

  background-color: transparent;
  cursor: pointer;

  transition: background-color .2s, color .2s;
}

.create-brand-buttons(@i: length(@brand-types)) when (@i > 0) {
  .create-brand-buttons(@i - 1);
  @type: extract(@brand-types, @i);
  @color: extract(@brand-colors, @i);

  .button-@{type}:extend(.button) {
    border-color: @color;

    &:hover {
      background-color: @color;
      color: #fff;
    }

    &:active {
      transform: translateY(1px);
    }

    &:focus {
      outline: none;
    }
  }
}

.create-brand-buttons();

.button-sized(@mod) {
  padding: @grid-base * @mod @grid-base * @mod * 2;
  font-size: 10px + @mod * 2;
}

.button-small {
  .button-sized(2);
}

.button-medium {
  .button-sized(3);
}

.button-large {
  .button-sized(4);
}

