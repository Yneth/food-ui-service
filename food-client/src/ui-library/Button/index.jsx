import React from 'react';
import PropTypes from 'prop-types';
import './style.less';

const Button = ({ type = 'primary', size = 'medium', onClick, children }) => {
    const className = `button-${type} button-${size}`;

    return <button className={className} onClick={onClick}>{children}</button>;
};

Button.propTypes = {
    children: PropTypes.string.isRequired,
    type: PropTypes.string,
    size: PropTypes.string,
    onClick: PropTypes.func,
};

export default Button;
