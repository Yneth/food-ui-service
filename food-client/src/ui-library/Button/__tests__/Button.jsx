import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '../index';

describe('Button component', () => {
    it('should render', () => {
        const tree = renderer.create(<Button>Hello</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with type specified', () => {
        const tree = renderer.create(<Button type="danger">Hello</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with size specified', () => {
        const tree = renderer.create(<Button size="large">Hello</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with custom className', () => {
        const tree = renderer.create(<Button className="custom_class">Hello</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call handler on click event', () => {
        const handler = jest.fn();
        const wrapper = mount(<Button onClick={handler}>Hello</Button>);
        wrapper.find('button').simulate('click');

        expect(handler).toHaveBeenCalled();
    });
});
