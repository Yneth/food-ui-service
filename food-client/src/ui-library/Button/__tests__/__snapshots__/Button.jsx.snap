// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
<button
  className="button-primary button-medium"
  onClick={undefined}
>
  Hello
</button>
`;

exports[`Button component should render with customization 1`] = `
<button
  className="button-danger button-large custom_class"
  onClick={undefined}
>
  Hello
</button>
`;
